service/src/test/java/uk/gov/hmcts/reform/fpl/handlers/judicial/NewHearingJudgeEventHandlerTest.java

```diff
@@ -189,4 +194,64 @@ void shouldHandleEditedHearingRolesWhenJudgeChanges() {
         verifyNoMoreInteractions(judicialService);
     }
 
+    @Test
+    void shouldCreateHearingRoleInFutureIfMultipleHearings() {
+        LocalDateTime future = LocalDateTime.now();
+
+        HearingBooking hearing1 = HearingBooking.builder()
+            .startDate(future)
+            .judgeAndLegalAdvisor(JudgeAndLegalAdvisor.builder()
+                .judgeJudicialUser(JudicialUser.builder()
+                    .idamId("idamId")
+                    .build())
+                .build())
+            .build();
+
+        HearingBooking hearing2 = hearing1.toBuilder()
+            .startDate(future.plusDays(2))
+            .build();
+
+        underTest.handleNewHearingJudge(new NewHearingJudgeEvent(
+            hearing2,
+            CaseData.builder().id(12345L).hearingDetails(wrapElements(hearing1, hearing2)).build(),
+            Optional.empty()
+        ));
+
+        verify(judicialService).assignHearingJudge(12345L,
+            "idamId",
+            future.plusDays(2).atZone(ZoneId.systemDefault()),
+            null,
+            false);
+    }
+
+    @Test
+    void shouldCreateHearingRoleNowIfSingleHearing() {
+        final LocalDateTime now = LocalDateTime.now();
+        final ZonedDateTime nowZoned = now.atZone(ZoneId.systemDefault());
+
+        HearingBooking hearing1 = HearingBooking.builder()
+            .startDate(now.plusDays(2))
+            .judgeAndLegalAdvisor(JudgeAndLegalAdvisor.builder()
+                .judgeJudicialUser(JudicialUser.builder()
+                    .idamId("idamId")
+                    .build())
+                .build())
+            .build();
+
+        try (MockedStatic<ZonedDateTime> zonedStatic = mockStatic(ZonedDateTime.class)) {
```

Comment:
Just wondered if you'd considered using a JudicialService Spy.

Reply:
Not too necessary here (unless I'm misunderstanding!), need to verify the exact dates the event handler is calling the service with, rather than checking the internal logic of the judicial service (tested elsewhere)